pkg/trace/remoteconfighandler/remote_config_handler.go

```diff
@@ -89,13 +91,56 @@ func (h *RemoteConfigHandler) Start() {
 		return
 	}
 
-	h.remoteClient.Start()
-	h.remoteClient.Subscribe(state.ProductAPMSampling, h.onUpdate)
-	h.remoteClient.Subscribe(state.ProductAgentConfig, h.onAgentConfigUpdate)
+	h.client.Start()
+	h.client.Subscribe(state.ProductAPMSampling, h.onUpdate)
+	h.client.Subscribe(state.ProductAgentConfig, h.onAgentConfigUpdate)
+	if h.mrfClient != nil {
+		h.mrfClient.Start()
+		h.mrfClient.Subscribe(state.ProductAgentFailover, h.mrfUpdateCallback)
+	}
+}
+
+// mrfUpdateCallback is the callback function for the AGENT_FAILOVER configs.
+// It fetches all the configs targeting the agent and applies the failover_apm settings.
+// Note: we only care about APM (failover_apm) configuration, but Logs (failover_logs) and Metrics (failover_metrics)
+// may also be present on the MRF configuration. These are handled on the core agent RC.
+func (h *RemoteConfigHandler) mrfUpdateCallback(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	var failoverAPM *bool
+	var failoverAPMCfgPth string
+	for cfgPath, update := range updates {
+		mrfUpdate, err := parseMultiRegionFailoverConfig(update.Config)
+		if err != nil {
+			pkglog.Errorf("Multi-Region Failover update unmarshal failed: %s", err)
+			applyStateCallback(cfgPath, state.ApplyStatus{
+				State: state.ApplyStateError,
+				Error: err.Error(),
+			})
+			continue
+		}
+
+		if mrfUpdate == nil || mrfUpdate.FailoverAPM == nil {
+			continue
+		}
+
+		if failoverAPM == nil || *mrfUpdate.FailoverAPM {
```

Comment:
Are we only updating the value if `*mrfUpdate.FailoverAPM` is true? Could it be the case that we want to update the agentconfig for MRF from true to false?

Reply:
We would also update the value if `mrfUpdate.Failover` is `false`, in case no other value has been assigned yet (first part of the conditional check). But if a value with `true` is received, that takes precedence. This is also the case with Metrics and Logs implementation of MRF.

Reply:
We rubber 🦆 over Slack and the code account for all possible updates 😄